klarna-on-demand/src/test/java/com/klarna/ondemand/PreferencesActivityTest.java

```diff
@@ -0,0 +1,94 @@
+package com.klarna.ondemand;
+
+import android.content.Intent;
+import android.view.MenuItem;
+import android.webkit.WebView;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Mockito;
+import org.mockito.configuration.DefaultMockitoConfiguration;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.rule.PowerMockRule;
+
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+import org.robolectric.tester.android.view.TestMenuItem;
+import org.robolectric.util.ActivityController;
+
+
+@RunWith(RobolectricTestRunner.class)
+@Config(emulateSdk = 18)
+@PrepareForTest(Context.class)
+@PowerMockIgnore({ "org.mockito.*", "org.robolectric.*", "android.*" })
+public class PreferencesActivityTest {
+
+    @Rule
+    public PowerMockRule rule = new PowerMockRule();
+
+    private PreferencesActivity preferencesActivity;
+    private ActivityController<PreferencesActivity> preferencesActivityController;
+
+    @Before
+    public void beforeEach() {
+        PowerMockito.mockStatic(Context.class);
+        Mockito.when(Context.getApiKey()).thenReturn("test_skadoo");
+
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.putExtra(preferencesActivity.EXTRA_USER_TOKEN, "my_token");
+        preferencesActivityController = Robolectric.buildActivity(PreferencesActivity.class).withIntent(intent).create();
+    }
+
+    @Test
+    public void handleUserErrorEvent_ShouldCallFinishWithRESULTERROR() {
```

Comment:
I could go for that.

On Wed, Jan 14, 2015 at 5:36 PM, tomerlevi444 <notifications@github.com>
wrote:

> In
> klarna-on-demand/src/test/java/com/klarna/ondemand/PreferencesActivityTest.java
> <https://github.com/klarna/klarna-on-demand-android/pull/2#discussion-diff-22943907>
> :
>
> > +
> > +    private PreferencesActivity preferencesActivity;
> > +    private ActivityController<PreferencesActivity> preferencesActivityController;
> > +
> > +    @Before
> > +    public void beforeEach() {
> > +        PowerMockito.mockStatic(Context.class);
> > +        Mockito.when(Context.getApiKey()).thenReturn("test_skadoo");
> > +
> > +        Intent intent = new Intent(Intent.ACTION_VIEW);
> > +        intent.putExtra(preferencesActivity.EXTRA_USER_TOKEN, "my_token");
> > +        preferencesActivityController = Robolectric.buildActivity(PreferencesActivity.class).withIntent(intent).create();
> > +    }
> > +
> > +    @Test
> > +    public void handleUserErrorEvent_ShouldCallFinishWithRESULTERROR() {
>
> @grozen <https://github.com/grozen> What about
> handleUserErrorEvent_shouldCallFinishWithResultError ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/klarna/klarna-on-demand-android/pull/2/files#r22943907>
> .
>



-- 
Die Aubergine ist purpurrot